src/components/Autocomplete/README.md

```diff
@@ -301,6 +302,128 @@ class AutocompleteExample extends React.Component {
 }
 ```
 
+### Autocomplete with lazy loading
+
+```jsx
+class AutoCompleteLazyLoadExample extends React.Component {
+  paginationInterval = 25;
+
+  options = Array.from(Array(1000)).map((_, index) => ({
+    value: `rustic ${index}`,
+    label: `Rustic ${index}`,
+  }));
+
+  state = {
+    selected: [],
+    inputText: '',
+    options: this.options,
+    visibleOptionIndex: this.paginationInterval,
+  };
+
+  render() {
+    const {visibleOptionIndex, selected, options} = this.state;
+    const textField = (
+      <Autocomplete.TextField
+        onChange={this.updateText}
+        label="Tags"
+        value={this.state.inputText}
+        placeholder="Vintage, cotton, summer"
+      />
+    );
+
+    const optionList = options.slice(0, visibleOptionIndex);
+    const selectedTagMarkup =
+      selected.length > 0 ? (
+        <Stack spacing="extraTight">{this.renderTags()}</Stack>
+      ) : null;
+
+    return (
+      <Stack vertical>
+        {selectedTagMarkup}
+        <Autocomplete
+          allowMultiple
+          options={optionList}
+          selected={this.state.selected}
+          textField={textField}
+          onSelect={this.updateSelection}
+          listTitle="Suggested Tags"
+          onLoadMoreResults={this.handleLoadMoreResults}
+        />
+      </Stack>
+    );
+  }
+
+  handleLoadMoreResults = () => {
+    const {visibleOptionIndex} = this.state;
+    const nextVisibleOptionIndex = visibleOptionIndex + this.paginationInterval;
+    if (nextVisibleOptionIndex <= this.options.length - 1) {
+      this.setState({visibleOptionIndex: nextVisibleOptionIndex});
+    }
+  };
+
+  updateText = (newValue) => {
+    this.setState({inputText: newValue});
+    this.filterAndUpdateOptions(newValue);
+  };
+
+  removeTag = (tag) => () => {
+    const newSelected = this.state.selected;
+    newSelected.splice(newSelected.indexOf(tag), 1);
```

Comment:
Or we could do `const {selected: newSeleced} = this.state;`, and `splice` would no longer be doing work directly on the state property, since destructuring creates something new.